src/utils/matter-parser.js

```diff
@@ -0,0 +1,98 @@
+const yaml = require('yaml');
+const fsp = require('fs').promises;
+const glob = require('glob');
+const { promisify } = require('util');
+
+/**
+ * Class for parsing a markdown file into frontmatter and content
+ */
+class MatterParser {
+  /**
+   * Creates a markdown file instance
+   * @param {string} fileName the file name
+   * @param {object} metaData the parsed frontmatter
+   * @param {string} content the content body
+   * @param {string} eol the line-ending style used (\n or \r\n)
+   */
+  constructor(fileName, metaData, content, eol = '\n') {
+    this.fileName = fileName;
+    this.metaData = metaData;
+    this.content = content;
+    this.eol = eol;
+  }
+
+  /**
+   * Read file
+   * @param {string} fileName the file name
+   * @returns {Promise<MatterParser>} the parser instance
+   */
+  static async fromFile(fileName) {
+    const content = await fsp.readFile(fileName, 'utf-8');
+    return MatterParser.fromString(content, fileName);
+  }
+
+  /**
+   * Read file from string
+   * @param {string} content contents of the frontmatter file
+   * @returns {MatterParser} the parser instance
+   */
+  static fromString(content = '', fileName = 'output.md') {
```

Comment:
The `fileName` argument is not documented in the JSDoc code block. :)

Reply:
thank you 👍🏻 :)